src/public/modules/forms/helpers/CsvMergedHeadersGenerator.ts

```diff
@@ -28,25 +45,25 @@ class CsvMergedHeadersGenerator extends CsvGenerator {
   /**
    * Returns current length of CSV file excluding header and meta-data
    */
-  length() {
+  length(): number {
     return this.unprocessed.length
   }
 
   /**
-   *
-   * @param {Object} decryptedContent
-   * @param {DisplayedResponse[]} decryptedContent.record
-   * @param {string} decryptedContent.created
-   * @param {string} decryptedContent.submissionId
+   * Extracts information from input record, rearranges record and then adds an UnprocessedRecord to this.unprocessed
+   * @param decryptedContent
+   * @param decryptedContent.record
+   * @param decryptedContent.created
+   * @param decryptedContent.submissionId
+   * @throws Error when trying to convert record into a response instance. Should be caught in submissions client factory.
    */
-  addRecord({ record, created, submissionId }) {
+  addRecord({ record, created, submissionId }: SubmissionRecord): void {
     // First pass, create object with { [fieldId]: question } from
     // decryptedContent to get all the questions.
     const fieldRecords = record.map((content) => {
       const fieldRecord = getResponseInstance(content)
```

Comment:
This won't fail with non-answer fields like Headers and Statements right? The old way returned a SingleAnswer record by default but now we throw an error.

Reply:
Yup I checked, it won't fail. They have `answer=""`, so should be fine!